internal/terraform/context_plan_ephemeral_test.go

```diff
@@ -549,8 +550,95 @@ You can correct this by removing references to ephemeral values, or by carefully
 				})
 			},
 		},
+
+		"write_only attribute": {
+			module: map[string]string{
+				"main.tf": `
+ephemeral "ephem_resource" "data" {
+}
+resource "ephem_write_only" "test" {
+    write_only = ephemeral.ephem_resource.data.value
+}
+`,
+			},
+			expectOpenEphemeralResourceCalled:           true,
+			expectValidateEphemeralResourceConfigCalled: true,
+			expectCloseEphemeralResourceCalled:          true,
+		},
+
+		"write_only with hardcoded value": {
+			toBeImplemented: true,
+			module: map[string]string{
+				"main.tf": `
+variable "ephem" {
+  type        = string
+  ephemeral   = true
+}
+
+resource "ephem_write_only" "producer" {
+    write_only = "this is not allowed"
+}
+
+output "out" {
+  value = ephemeralasnull(var.ephem)
+}
+`,
+			},
+			inputs: InputValues{
+				"ephem": &InputValue{
+					Value: cty.StringVal("ami"),
+				},
+			},
+			expectPlanDiagnostics: func(m *configs.Config) (diags tfdiags.Diagnostics) {
+				return diags.Append(&hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Cannot use hard-coded value in write-only attribute",
+					Detail:   `The attribute "write_only" of resource "ephem_write_only.producer" is write-only and cannot be used with a static value, please use an ephemeral variable or an ephemeral resource instead.`,
+				})
+			},
+		},
+
+		"write_only attribute references": {
+			toBeImplemented: true,
+			module: map[string]string{
+				"main.tf": `
+variable "ephem" {
+  type        = string
+  ephemeral   = true
+}
+
+resource "ephem_write_only" "producer" {
+    write_only = var.ephem # Valid usage
+}
+
+resource "ephem_write_only" "consumer" {
+    write_only = ephem_write_only.producer.write_only # Invalid usage
```

Comment:
I would leave this to a possible later PR about validation. This isn't really a validation that is possible, but we could decide that detecting some direct references is OK.

Reply:
ok 👍